ddtrace/appsec/_remoteconfiguration.py

```diff
@@ -86,8 +86,9 @@ def _add_rules_to_list(features: Mapping[str, Any], feature: str, message: str,
     if rules is not None:
         try:
             if ruleset.get(feature) is None:
-                ruleset[feature] = []
-            ruleset[feature] += rules
+                ruleset[feature] = rules
+            else:
+                ruleset[feature] = ruleset[feature] + rules
```

Comment:
This would be equivalent I think, although maybe too condensed?
```suggestion
            ruleset.setdefault(feature, []).extend(rules)
```

Reply:
not equivalent, because the initial list would be modified, and I don't want that if feature is initialized from the default rule set. So this code was made to ensure immutability. We rebuild a list if we have to append something.